File: app/src/main/java/com/popalay/barnee/ui/screen/discovery/DiscoveryViewModel.kt
package com.popalay.barnee.ui.screen.discovery

import com.popalay.barnee.domain.discovery.DiscoveryAction
import com.popalay.barnee.domain.discovery.DiscoveryState
import com.popalay.barnee.domain.discovery.DiscoveryStateMachine
import com.popalay.barnee.ui.screen.StateMachineWrapperViewModel

class DiscoveryViewModel(
    stateMachine: DiscoveryStateMachine
) : StateMachineWrapperViewModel<DiscoveryState, DiscoveryAction>(stateMachine)

File: ios/iosApp/Screen/Discovery/DiscoveryViewModel.swift
//
//  DiscoveryViewModel.swift
//  iosApp
//
//  Created by Denys Nykyforov on 25/3/21.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import shared
import SwiftUI

class DiscoveryViewModel: StateMachineWrapperViewModel<DiscoveryState, DiscoveryMutation, EmptySideEffect, DiscoveryStateMachine> {
    convenience init() {
        let sharedComponent = (UIApplication.shared.delegate as! AppDelegate).sharedComponent
        self.init(stateMachine: sharedComponent.provideDiscoveryStateMachine(), initialState: DiscoveryState(categories: Uninitialized()))
    }
}
